packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt

```diff
@@ -0,0 +1,467 @@
+      .groupBy { it.parentFile }
+
+    if (albumsMap.size != 1) {
+      // Empty albums shouldn't be visible to users. That's why this is an error.
+      promise.reject(ERROR_NO_ALBUM, "Found album is empty.")
+      return
+    }
+    val albumDir = assets[0].parentFile
+    if (albumDir == null) {
+      promise.reject(ERROR_NO_ALBUM, "Couldn't get album path.")
+      return
+    }
+    if (albumDir.canWrite()) {
+      // Nothing to migrate
+      promise.resolve(null)
+      return
+    }
+
+    val action = Action action@{ permissionsWereGranted: Boolean ->
+      if (!permissionsWereGranted) {
+        promise.reject(ERROR_NO_PERMISSIONS, ERROR_USER_DID_NOT_GRANT_WRITE_PERMISSIONS_MESSAGE)
+        return@action
+      }
+      MigrateAlbum(mContext, assets, albumDir.name, promise)
+        .executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
+    }
+    val needsToCheckPermissions = assets.map { it.assetId }
+    runActionWithPermissions(needsToCheckPermissions, action, promise)
+  }
+
+  @ExpoMethod
+  fun albumNeedsMigrationAsync(albumId: String, promise: Promise) {
+    if (isMissingPermissions) {
+      promise.reject(ERROR_NO_PERMISSIONS, ERROR_NO_PERMISSIONS_MESSAGE)
+      return
+    }
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+      CheckIfAlbumShouldBeMigrated(mContext, albumId, promise)
+        .executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
+      return
+    }
+    promise.resolve(false)
+  }
+
+  // Library change observer
+  @ExpoMethod
+  fun startObserving(promise: Promise) {
+    if (mImagesObserver != null) {
+      promise.resolve(null)
+      return
+    }
+
+    // We need to register an observer for each type of assets,
+    // because it seems that observing a parent directory (EXTERNAL_CONTENT) doesn't work well,
+    // whereas observing directory of images or videos works fine.
+    val handler = Handler()
+    val contentResolver = mContext.contentResolver
+
+    mImagesObserver =
+      MediaStoreContentObserver(handler, MediaStore.Files.FileColumns.MEDIA_TYPE_IMAGE)
+        .also { imagesObserver ->
+          contentResolver.registerContentObserver(
+            MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+            true,
+            imagesObserver
+          )
+        }
+
+    mVideosObserver =
+      MediaStoreContentObserver(handler, MediaStore.Files.FileColumns.MEDIA_TYPE_VIDEO)
+        .also { videosObserver ->
+          contentResolver.registerContentObserver(
+            MediaStore.Video.Media.EXTERNAL_CONTENT_URI,
+            true,
+            videosObserver
+          )
+        }
+
+    promise.resolve(null)
+  }
+
+  @ExpoMethod
+  fun stopObserving(promise: Promise) {
+    if (mImagesObserver != null) {
+      val contentResolver = mContext.contentResolver
+      contentResolver.unregisterContentObserver(mImagesObserver!!)
+      contentResolver.unregisterContentObserver(mVideosObserver!!)
+      mImagesObserver = null
+      mVideosObserver = null
+    }
```

Comment:
```suggestion
    val contentResolver = mContext.contentResolver
    mImagesObserver?.let {
      contentResolver.unregisterContentObserver(it)
      mImagesObserver = null
    }
    mVideosObserver.?let {
     contentResolver.unregisterContentObserver(it)
     mVideosObserver = null
    }
```
It's a little bit safer 😉